src/eval_env.cc

```diff
@@ -107,12 +107,7 @@ string EvalString::Evaluate(Env* env) const {
 }
 
 void EvalString::AddText(StringPiece text) {
-  // Add it to the end of an existing RAW token if possible.
-  if (!parsed_.empty() && parsed_.back().second == RAW) {
```

Comment:
Why is this change safe?

I think a comment indicating the change in semantics, and justification thereof, of the AddText function would be wise here.